src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -544,6 +547,36 @@ calculateFloatingPaymentClaimsFromSwapStream floatingRateCalculation s
             where
               rate = getFloatingRate s pp.calculationPeriod p regularRate rateFixingCalendar fixLeg
 
+      -- calculate simple interest rate claim (no calc period compounding) for a payment period
+      -- using a compounded reference rate
+      createSimpleClaimCompoundedRefRate regularRate p =
+        mconcat $ map createSimpleRateCompoundingRefRate paymentPeriodData
+        where
+          (notional, paymentPeriodData) = calculateNotionalAndFixings p s
+            calculationScheduleStaticData fxRateId useAdjustedDatesForDcf periodicSchedule
+          createSimpleRateCompoundingRefRate pp = mconcat $ [claim, correctionClaim]
+            where
+              rate = getFloatingRate s pp.calculationPeriod p regularRate rateFixingCalendar fixLeg
+                -- Act360 was implicitly used for the SOFR Index, see page 35 here:
+                -- https://www.newyorkfed.org/medialibrary/Microsites/arrc/files/2021/users-guide-to-sofr2021-update.pdf
+              dcfMethodImpliedInSofrIndex = Act360
+              dcfImpl = Const (calcPeriodDcf dcfMethodImpliedInSofrIndex
+                pp.calculationPeriod useAdjustedDatesForDcf periodicSchedule.terminationDate
+                periodicSchedule.frequency)
+              notionalAndDcf = if pp.rateFixingDate < pp.calculationPeriod.adjustedEndDate then
+                error "fixing of notional must not occur before the final calculation date"
+                else
+                  when (TimeGte pp.rateFixingDate) $ scale notional $
+                  scale (pp.dcf / dcfImpl) $
```

Comment:
Not fully sure what is happening here, could you help me understand it?

Reply:
The formula Compound Annualized Interest on p36 
https://www.newyorkfed.org/medialibrary/Microsites/arrc/files/2021/users-guide-to-sofr2021-update.pdf
contains the factor `360/d_c`, which corresponds to `1 / dcfImpl` (using the **Act360** convention, as described on p35).

That gives you the p.a. rate that should be used for this swap. However, for this swap we also need to use the swap-specific daycount convention `pp.dcf`.

Reply:
Got it, so we are basically calculating `I_y / I_x` and then annualising it using ` 1 / impliedDcf`.

I think I got confused by the variable name `notionalAndDcf`, which includes `notional * dcf / implied dcf`. I would rather define an `annualisedRate = compounded rate / implied DCF` and then an `effectiveRate = annualise Rate * DCF`. 

As a side note, I'd be keen to understand why the following sentence holds : Because the SOFR Index implements the ISDA compound SOFR definition, it can be used to
calculate accrued interest in a plain arrears convention and also in a convention with compound
interest and lookback with observation shift. However, the Index cannot be used with other
conventions, such as a lookback without shift or a lockout

We could try to figure that out in the office on Wed.

Reply:
> I think I got confused by the variable name `notionalAndDcf`, which includes `notional * dcf / implied dcf`. I would rather define an `annualisedRate = compounded rate / implied DCF` and then an `effectiveRate = annualise Rate * DCF`.

I don't know if that works since we have to submit `rate` to `scaleByAssetPerformance`. I have renamed the variables and added an extra comment to explain the calculation.